src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -320,6 +337,10 @@ private async Task DeleteTopicSubscriptionAsync(CancellationToken cancellationTo
                     Logger.LogTrace("Cannot delete topic subscription with name '{SubscriptionName}' because no subscription exists on Service Bus resource", SubscriptionName);
                 }
             }
+            catch (Exception exception) when (exception is TaskCanceledException || exception is OperationCanceledException)
+            {
+                // Ignore.
```

Comment:
Same remark regarding the exception swallowing :)

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/313/commits/39abf09b93a637870cc5542689716bfc1242aa54